### DISCUSSION TITLE
Testcase2 Error

### DISCUSSION THREAD

**Discussion by Lise-Li** [NASA-LIS/LISF#1434] [Q&A] [answered]
Hi, LISF Community.
I'm running the public testcase walkthrough, and I met the following error when I run the step2. I have never changed the lis.config, and download the NLDAS data through the  wget.sh provieded by the INPUT directory. I provide my lislog_ol.0000 and lis.config in the attached files. Please find the attached files.
[lislog_ol.0000.txt](https://github.com/NASA-LIS/LISF/files/13052215/lislog_ol.0000.txt)
[lis.config_noah36_ol.txt](https://github.com/NASA-LIS/LISF/files/13052235/lis.config_noah36_ol.txt)

Please kindly let me know how to solve this error.
Thank you!
Lise Li

**Comment by dmocko** [comment]
Hello Lise Li,

From your running directory (where the `LIS` executable is found), do you see this file if you do this?
   ` ls -l ./INPUT/NLDAS2.FORCING/2017/001/NLDAS_FORA0125_H.A20170101.0000.002.grb`

Perhaps the NLDAS-2 forcing files were downloaded into the wrong directory.  Have you followed all the steps in the "Download the Meteorological Forcing Dataset" section of:
    https://nasa-lis.github.io/LISF/public_testcase_walkthrough/public_testcase_walkthrough.html#_step_2_lsm_open_loop_ol_experiment_lis

Also, are you using GRIBAPI or eccodes for your GRIB library?

David

**Reply by Lise-Li** [reply]
Hello, @dmocko! 
Thank you for your reply! 
I have checked the NLDAS-2 forcing directory, and it is in the right directory. Also, I'm sure that I download the forcing data followed all the steps in step2. I checked the downloaded data using panoply and the data covers the US, so I don't think it should be a data issue. I confirmed that my eccodes had been installed successfully by typing codes_info into my command line. But I'm not sure if I used eccodes. About eccodes I set up the LIS_ECCODES environment when configuring LIS, and when setting up for LIS, Use GRIBAPI/ECCODES? (0-neither, 1-gribapi, 2-ECCodes, default=2) I chose eccodes. I'm not sure if this is using eccodes for GRIB library?

Looking forward to your reply!
Thank you!
Lise Li

**Reply by dmocko** [reply]
Hello,

Please attach your configure.lis file (in LISF/lis/make) as well as the result of `echo $LD_LIBRARY_PATH`.

**Reply by Lise-Li** [reply]
Hello, @dmocko !

Thank you for your reply! The following are my configure.lis and result of echo $LD_LIBRARY_PATH.
[configure.lis.txt](https://github.com/NASA-LIS/LISF/files/13097241/configure.lis.txt)
![result of echo $LD_LIBRARY_PATH](https://github.com/NASA-LIS/LISF/assets/145978291/2b496dde-17a7-4dbf-945c-835c9182a565)


Please kindly let me know how to solve this error.
Thank you!
Lise Li

**Reply by dmocko** [reply]
Hi - I don't see your entire LD_LIBRARY_PATH.  Can you try again please?

`echo $LD_LIBRARY_PATH > ld_library_path.txt`

And attach the .txt file.  Do you have the eccodes lib in your LD_LIBRARY PATH?

**Reply by dmocko** [reply]
Also please attach your `./INPUT/forcing_variables.txt` file.

And I will ask more directly this time.  Please copy exactly what you see when you type this in the same directory as the LIS executable:

`ls -l ./INPUT/NLDAS2.FORCING/2017/001/NLDAS_FORA0125_H.A20170101.0000.002.grb`

The more I look into this, it seems to be an issue with you accessing this file (permissions?) or the download was corrupted.

**Reply by Lise-Li** [reply, marked as answer]
Hi, @dmocko 

This morning after your inspiration, I realized that my problem was that I didn't set the eccodes library path in ~/.bashrc, so I set it and regenerated the LIS executable. Currently, step2 is running successfully. Thanks again for your help! Also attached are a couple of files you mentioned to make sure I did it right this time.
[configure.lis.txt](https://github.com/NASA-LIS/LISF/files/13116527/configure.lis.txt)
[ld_library_path.txt](https://github.com/NASA-LIS/LISF/files/13116060/ld_library_path.txt)
[forcing_variables.txt](https://github.com/NASA-LIS/LISF/files/13116065/forcing_variables.txt)
The following is after I type the ls command:
-rw-rw-r-- 1 lihuijie lihuijie 1732574 Feb 16  2018 ./INPUT/NLDAS2.FORCING/2017/001/NLDAS_FORA0125_H.A20170101.0000.002.grb

Please kindly let me know if I did it right this time.
Thank you 
Lise Li

**Reply by dmocko** [reply]
Glad that it is resolved.  Yes, that's correct, and the file size is accurate for that file.

**Reply by Lise-Li** [reply]
Greate! Many thanks for your help!
